src/app/services/baw-api/baw-api.service.ts

```diff
@@ -208,23 +217,58 @@ export class BawApiService<Model extends AbstractModel> {
    * @param body Request body
    */
   protected apiCreate(path: string, body: AbstractModel): Observable<Model> {
-    return this.httpPost(path, body.toJSON?.({ create: true }) ?? body).pipe(
+    const jsonData = body.toJSON?.({ create: true });
+    return this.httpPost(path, jsonData ?? body).pipe(
       map(this.handleSingleResponse)
     );
   }
 
   /**
-   * Get response from update route
+   * Get response from create route with an additional multipart body update
+   * request
    *
-   * @param path API path
+   * @param createPath API create path
+   * @param updatePath API update path
    * @param body Request body
    */
-  protected apiUpdate(path: string, body: AbstractModel): Observable<Model> {
-    return this.httpPatch(path, body.toJSON?.({ update: true }) ?? body).pipe(
+  protected apiCreateMultipart(
+    createPath: string,
+    updatePath: (model: Model) => string,
+    body: AbstractModel
+  ) {
+    const formData = body.toFormData({ create: true });
+    return this.apiCreate(createPath, body).pipe(
+      mergeMap((model) =>
+        this.httpPut(updatePath(model), formData, multiPartHeaders)
+      ),
       map(this.handleSingleResponse)
     );
   }
 
+  /**
+   * Get response from update route. If the model has form data, this will make
+   * an additional multipart update request.
+   *
+   * @param path API path
+   * @param body Request body
+   */
+  protected apiUpdate(path: string, body: AbstractModel): Observable<Model> {
+    const jsonData = body.toJSON?.({ update: true });
+
+    if (body.hasFormData({ update: true })) {
+      const formData = body.toFormData?.({ update: true });
+      return this.httpPatch(path, jsonData ?? body).pipe(
+        map(this.handleSingleResponse),
+        mergeMap(() => this.httpPut(path, formData, multiPartHeaders)),
+        map(this.handleSingleResponse)
+      );
+    } else {
+      return this.httpPatch(path, jsonData ?? body).pipe(
+        map(this.handleSingleResponse)
+      );
```

Comment:
this can probably be flattened into one chain

Reply:
Tried to, but I'm unsure of a good method to maintain type support. Potentially a const array, or maybe a function

Reply:
Nvm, figured out how